appium/lib/screen-object/accessors/element.rb

```diff
@@ -37,88 +34,179 @@ def driver
       end
 
       def click
-          element.click
+        element.click
       end
 
       def value
-          element.value
+        element.value
       end
 
       def exists?
+        driver.no_wait
         begin
-          element.displayed?
-        rescue
+          driver.set_wait if element.displayed?
+          true
+        rescue StandardError
           false
         end
       end
 
       def element
-        driver.find_element(:"#{locator[0]}",locator[1])
+        driver.find_element(:"#{locator[0]}", locator[1].to_s)
       end
 
       def elements
-        driver.find_elements(:"#{locator[0]}",locator[1])
+        driver.find_elements(:"#{locator[0]}", locator[1].to_s)
       end
 
       def element_attributes
         %w[name resource-id value text]
       end
 
+      def rect
+        my_element = element.rect
+        {
+            start_x: my_element.x,
+            end_x: my_element.x + my_element.width,
+            start_y: my_element.y,
+            end_y: my_element.y + my_element.height,
+            height: my_element.height,
+            width: my_element.width
+        }
+      rescue RuntimeError => err
+        raise("Error Details: #{err}")
+      end
+
       def dynamic_xpath(text)
-        concat_attribute=[]
-        element_attributes.each{|i| concat_attribute << %Q(contains(@#{i}, '#{text}'))}
-        puts  "//#{locator[0]}[#{concat_attribute.join(' or ')}]"
-        locator1="xpath~//#{locator[0]}[#{concat_attribute.join(' or ')}]"
-        @locator=locator1.split("~")
+        concat_attribute = []
+        element_attributes.each { |i| concat_attribute << %(contains(@#{i}, '#{text}')) }
+        puts "//#{locator[0]}[#{concat_attribute.join(' or ')}]"
+        locator1 = "xpath://#{locator[0]}[#{concat_attribute.join(' or ')}]"
```

Comment:
Again, this is only if you send locator as a string, rather than changing the separator from ~ to : just convert it to a `Hash` i.e.
```
locator1 = { xpath: "//#{locator[0]}[#{concat_attribute.join(' or ')}]" }
@locator = locator1
```